### ISSUE TITLE
[Q] Restriction to lidar devices?

### ISSUE THREAD

**Issue by PulsarFX** [Tasshack/dreame-vacuum#12]
Does this component rely on capabilities of lidar devices?
I have a dreame F9 which has no lidar tower but a camera for navigation. Would it be supported by this component? 
F9 can do all the things a lidar device can do, too: map support, section and room cleaning.

Thanks for your help.

**Comment by Zwaubel** [comment]
@Tasshack 
Can you by any chance also include the Xiaomi Vacuum Robot 1C?
The model ID is most probably `dreame.vacuum.p1904`.

**Comment by fendirce** [comment]
> 

Hi @Tasshack ,

When will List of Vslam robots be supported?
Waiting Impatiently
thanks for the quick turnaround

**Comment by Zwaubel** [comment]
@Tasshack Okay, that is a bummer. I was not aware of these differences. 
Thanks for linking the related issue though. I will take a closer  look at your code and the API. Maybe I will come up with something to contribute.

**Comment by SandDeath** [comment]
> 

When you get to the tests, I can not only grant access to the device, but full access to the account.  We are looking forward to the implementation, now I am busy adding a translation into Russian

**Comment by Zwaubel** [comment]
@Tasshack 
Regarding map saving: If I deactivate that particular option from within Xiaomi Home, the robot recreates the map every time I start a cleaning. So probably the map really is not being saved permanently.  

How do you extract the map data from the robot? I can provide you with new data, if you lead me the way. Would also help myself regarding the integration of the Mijia 1C.

**Comment by SandDeath** [comment]
> @SandDeath i am going to start development for the VSLAM support if you give me access to your device through Mi Home app.

of course, how can I contact to provide the username / password from the account, or give me your MiHome account ID so that I can grant access to the device

**Comment by Zwaubel** [comment]
> @Zwaubel 
> 
> > Regarding map saving: If I deactivate that particular option from within Xiaomi Home, the robot recreates the map every time I start a cleaning. So probably the map really is not being saved permanently.
> 
> 
> 
> How does map look like when you disable map saving before start cleaning. Can you share ascreenshot?
> 
> 
> 
> > How do you extract the map data from the robot? I can provide you with new data, if you lead me the way. Would also help myself regarding the integration of the Mijia 1C.
> 
> 
> 
> Actually, extracting the map data from the robot is not possible instead it can only be acquired from the cloud database. It is a little complicated process but you can checkout how it is done from the [Xiaomi Cloud Map Extractor](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor/blob/master/custom_components/xiaomi_cloud_map_extractor/common/vacuum_v2.py) source code.
> 
> 
> 
> 

When starting with map saving disabled, the robot starts within a black void. The map than starts to build progressively the further the robot is driving. 
![image](https://user-images.githubusercontent.com/25179994/200681351-4dde0380-bab1-4369-829d-0ac856e853af.jpeg)
This process looks very similar to the initial map when map saving is turned on. 
![image](https://user-images.githubusercontent.com/25179994/200681936-44a8cf9d-f6a7-4281-a769-fe26d858d796.jpeg)

I'll try to Look at the map extractor this weekend.

**Comment by alexdelli** [comment]
> Support for VSLAM robots is almost ready and will be released soon. Thanks to @SandDeath for giving me access to his device, without it this cannot be possible.

Thank you Tasshack and thanks to @SandDeath for sure!!

**Comment by SandDeath** [comment]
> I was planning to release it last month, it still needed some testing but unfortunately the battery of device that i have been using for testing has died.

My new battery is on its way, AliExpress promises delivery by the end of the month and if necessary we can continue testing

**Comment by alexdelli** [comment]
I can offer my robot for testing if needed. Is it not necessary to change firmware ?

**Comment by ezequiel-g199** [comment]
I also have a VSLAM device if you need it for more testing.

**Comment by michalkruczek** [comment]
Hi do we have any updates ? I'm also interested in implementing my p2041o (mi vacuum mop pro 2 ) into HA with map features. Btw Many thanks for Your support @Tasshack .

**Comment by humanBeer** [comment]
No idea. Im also waiting for an update for p2148o.

The standard integration that didn't update the map now suddenly doesn't work at all anymore, and I'm not bothered to fix it if this will hopefully soon come out.

**Comment by cnecrea** [comment]
Hi there,

Any updates regarding dreame.vacuum.p2041o (Mi Robot Vacuum-Mop 2 Pro+)?
I can share my vacuum in order to get this update/fix.

**Comment by foxel** [comment]
Hi guys. Any update on this?

@Tasshack do you need help? 

BTW. I don't see any in-progress branches here

**Comment by benavern** [comment]
Hi,
I've tried to use my dreame f9 with this integration and it works like a charm without the map. Did someone try with it ? At least I can integrate it in my automations ...

**Comment by esvanegas** [comment]
Hi everyone!

@Tasshack any update on this? do you need any help?

**Comment by szwarx** [comment]
Please add support for xiaomi vacuum-mop 2 ijai.vacuum.v3

**Comment by aristizabal95** [comment]
Hello! Chiming in to say I'd also love to see this implemented, and I can offer access to my device (`dreame.vacuum.p2140o`) and account if needed for testing.

**Comment by theonlysinjin** [comment]
This is amazing thank you so much!
I updated the addon and hit the add button under the [configure section](https://github.com/Tasshack/dreame-vacuum#configuration) and like magic it works!

**Comment by aristizabal95** [comment]
Works like a charm! This made my day 🥳

**Comment by michalkruczek** [comment]
Hi @Tasshack , I’ve got the same issue as person above. Full name of my vacuum is “ Xiaomi Mi Vacuum-Mop Pro 2 ”

**Comment by djevrek** [comment]
Yea, same here, same device and same error.  Full spec name is ijai.vacuum.v3

**Comment by shefpsk** [comment]
Everything works for me (dreame.vacuum.p2140o)!!!! I'm happy! @Tasshack thank you!!! You're great!

**Comment by AndreyMartjushev** [comment]
hello, p2140p connected in the first version (it was possible through ip and token, no devices were found through the device account), then the integration was updated to version 2... the card did not appear, I deleted everything and decided to connect again. after that, my vacuum cleaner stopped connecting at all.
vacuum mop 2
Are there any solutions?

**Comment by AndreyMartjushev** [comment]
@Tasshack yes, I managed to return the version to 1, but there again without a map (((

Tell me, is there any way to clean a certain room without camera support? zone cleaning
